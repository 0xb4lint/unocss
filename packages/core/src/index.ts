import { createGenerator as _createGenerator } from './generator'
import { resolveConfig as _resolveConfig } from './config'
import { UserConfigDefaults } from './types'

export * from './types'
export * from './utils'
export * from './extractors'

export type ArgumentType<T> = T extends ((...args: infer A) => any) ? A : never
export type Shift<T> = T extends [_: any, ...args: infer A] ? A : never
export type RestArgs<T> = Shift<ArgumentType<T>>

export function createMiniwindWithDefaults(defaults: UserConfigDefaults) {
  return {
    createGenerator: (...args: RestArgs<typeof _createGenerator>) => _createGenerator(defaults, ...args),
    resolveConfig: (...args: RestArgs<typeof _resolveConfig>) => _resolveConfig(defaults, ...args),
  }
}
