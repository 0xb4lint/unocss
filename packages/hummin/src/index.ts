export * from '@hummin/core'
export * from '@hummin/preset-default'
export * from '@hummin/preset-attributify'
