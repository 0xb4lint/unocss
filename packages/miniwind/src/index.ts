export * from '@miniwind/core'
export * from '@miniwind/preset-default'
export * from '@miniwind/preset-attributify'
