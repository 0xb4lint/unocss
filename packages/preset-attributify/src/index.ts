import { Preset } from '@miniwind/core'
import { extractorAttributify } from './extractor'
import { AttributifyOptions } from './types'
import { variantAttributify } from './variant'

export * from './extractor'
export * from './variant'
export * from './types'

export const presetAttributify = (options?: AttributifyOptions): Preset => ({
  variants: [
    variantAttributify(options),
  ],
  extractors: [
    extractorAttributify,
  ],
})
