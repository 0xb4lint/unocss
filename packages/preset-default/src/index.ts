import { createMiniwindWithDefaults, extractorSplit, Preset } from '@miniwind/core'
import { defaultRules } from './rules'
import { defaultTheme } from './theme'
import { defaultVariants } from './variants'

export * from './rules'
export * from './variants'
export * from './theme'

export const presetDefault = (): Preset => ({
  rules: defaultRules,
  variants: defaultVariants,
  extractors: [extractorSplit],
})

export const {
  createGenerator,
  resolveConfig,
} = createMiniwindWithDefaults({
  theme: defaultTheme,
  presets: [
    presetDefault(),
  ],
})
