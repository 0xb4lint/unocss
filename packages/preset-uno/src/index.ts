import { extractorSplit, Preset } from '@unocss/core'
import { rules } from './rules'
import { variants } from './variants'

export const preset = (): Preset => ({
  rules,
  variants,
  extractors: [extractorSplit],
})

export default preset
