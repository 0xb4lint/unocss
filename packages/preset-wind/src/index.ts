import { Preset } from '@unocss/core'

export const preset = (): Preset => ({
  // TODO:
})

export default preset
