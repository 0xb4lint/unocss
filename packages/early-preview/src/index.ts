import { createHash } from 'crypto'
import { UserConfig } from '@miniwind/core'
import { presetAttributify } from '@miniwind/preset-attributify'
import { presetDefault } from '@miniwind/preset-default'
import plugin from '@miniwind/vite'

function getHash(input: string, length = 8) {
  return createHash('sha256')
    .update(input)
    .digest('hex')
    .substr(0, length)
}

export default (obj: UserConfig, key: string) => {
  // 0dc2327d
  if (key !== getHash('mwmwmw'))
    return null

  return plugin({
    presets: [
      presetAttributify(),
      presetDefault(),
    ],
    ...obj,
  })
}
