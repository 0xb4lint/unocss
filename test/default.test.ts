import fs from 'fs'
import { createGenerator } from '../src'

const fixture = [
  'p-1',
  'pt-2',
  'p-3',
  'pl-10px',
  'hover:p-4',
  '!p-5px',
  '!hover:px-10',
  'hover:!p-10',
  'op-10',
  'opacity-0',
  'flex',
  'text-red-300',
  'text-blue',
  'text-warn-gray',
  'text-black/10',
  'bg-teal-100/55',
  'border',
  'border-2',
  'font-mono',
  'm-auto',
  'my-auto',
  'm-[3em]',
  'sm:text-red-100',
  'sm:text-red-200',
  'sm:text-red-300',
  'md:!hidden',
  'rounded-t-sm',
  'rounded-md',
  'rounded-tr',
  'rounded-1/2',
  'rounded-rb-1/2',
  'rounded-full',
  'rounded-[4px]',
  'rounded',
  'dark:not-odd:text-red',
  'hover:not-first:checked:bg-red/10',
  'text-base',
  'text-lg',
  'text-4xl',
  'bg-red-100 bg-opacity-45',
  'border-b border-t-2 border-green-100/10',
].join(' ')

const generator = createGenerator()

test('default', async() => {
  expect(await generator(fixture)).toMatchSnapshot()
})

test('scope', async() => {
  expect(await generator(fixture, '', '.foo-scope')).toMatchSnapshot()
})
