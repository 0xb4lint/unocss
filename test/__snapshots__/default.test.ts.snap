// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
"
.p-2, .p2{padding:0.5rem;}
.hover\\\\:p-4:hover{padding:1rem;}
.\\\\!p-5px{padding:5px !important;}
.hover\\\\:\\\\!p-10:hover{padding:2.5rem !important;}
.\\\\!hover\\\\:px-10:hover{padding-left:2.5rem !important;padding-right:2.5rem !important;}


.pt-2, .p-t-2, .pt2{padding-top:0.5rem;}
.pl-10px{padding-left:10px;}
.m-auto{margin:auto;}
.m-\\\\[3em\\\\]{margin:3em;}
.my-auto{margin-top:auto;margin-bottom:auto;}
.op-10{opacity:0.1;}
.opacity-0{opacity:0;}
.bg-teal-100\\\\/55{background-color:rgba(204,251,241,0.55);}
.hover\\\\:not-first\\\\:checked\\\\:bg-red\\\\/10:hover:not(:first):checked{background-color:rgba(248,113,113,0.1);}
.bg-red-100{--mw-bg-opacity:1;background-color:rgba(254,226,226,var(--mw-bg-opacity));}
.bg-\\\\[\\\\#153\\\\]\\\\/10{background-color:rgba(17,85,51,0.1);}

.bg-\\\\#452233\\\\/40, .bg-hex-452233\\\\/40{background-color:rgba(69,34,51,0.4);}
.bg-opacity-45{--mw-bg-opacity:0.45;}
.border{border-width:1px;border-style:solid;}
.border-2{border-width:2px;border-style:solid;}
.border-b{border-bottom-width:1px;border-style:solid;}
.border-t-2{border-top-width:2px;border-style:solid;}
.border-green-100\\\\/10{border-color:rgba(220,252,231,0.1);}
.rounded-t-sm{border-top-left-radius:0.125rem;border-top-right-radius:0.125rem;}
.rounded-md{border-radius:0.375rem;}
.rounded-tr{border-top-right-radius:0.25rem;}
.rounded-1\\\\/2{border-radius:50%;}
.rounded-rb-1\\\\/2{border-bottom-right-radius:50%;}
.rounded-full{border-radius:9999px;}
.rounded-\\\\[4px\\\\]{border-radius:4px;}
.rounded{border-radius:0.25rem;}
.font-mono{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace;}
.text-base{font-size:1rem;line-height:1.5rem;}
.text-lg{font-size:1.125rem;line-height:1.75rem;}
.text-4xl{font-size:2.25rem;line-height:2.5rem;}
.text-red-300{--mw-text-opacity:1;color:rgba(252,165,165,var(--mw-text-opacity));}
.text-blue{--mw-text-opacity:1;color:rgba(96,165,250,var(--mw-text-opacity));}
.text-black\\\\/10{color:rgba(0,0,0,0.1);}
.dark .dark\\\\:not-odd\\\\:text-red:not(:nth-child(odd)){--mw-text-opacity:1;color:rgba(248,113,113,var(--mw-text-opacity));}
.text-\\\\[\\\\#124\\\\]{--mw-text-opacity:1;color:rgba(17,34,68,var(--mw-text-opacity));}
.flex{display:flex;}
.order-first{order:-9999;}
.z-100{z-index:100;}
.blur-4{--mw-blur:blur(4px);filter:var(--mw-blur) var(--mw-brightness) var(--mw-contrast) var(--mw-grayscale) var(--mw-hue-rotate) var(--mw-invert) var(--mw-saturate) var(--mw-sepia) var(--mw-drop-shadow);}
@media (min-width: 640px){
.sm\\\\:text-red-100{--mw-text-opacity:1;color:rgba(254,226,226,var(--mw-text-opacity));}
.sm\\\\:text-red-200\\\\/10{color:rgba(254,202,202,0.1);}
.sm\\\\:text-red-300\\\\/20{color:rgba(252,165,165,0.2);}
}
@media (min-width: 768px){
.md\\\\:\\\\!hidden{display:none !important;}
}"
`;
